src/KeyControls.ts

```diff
@@ -0,0 +1,59 @@
+class KeyControls {
+  keys: {};
+  
+  constructor() {
+    this.keys = {};
+
+    // Bind event handlers
+    document.addEventListener("keydown", e => {
+      if ([37, 38, 39, 40, 32].indexOf(e.which) >= 0) {
+        e.preventDefault();
+      }
+      this.keys[e.which] = true;
+    }, false);
+
+    document.addEventListener("keyup", e => {
+      this.keys[e.which] = false;
+    }, false);
+  }
+
+  key(key, value) {
+    if (value !== undefined) {
+      this.keys[key] = value;
+    }
+    return this.keys[key];
+  }
+
+  reset () {
+    for (let key in this.keys) {
+      this.keys[key] = false;
+    }
+  }
+
+  // Handle key actions
+  get action() {
+    return this.keys[32];
+  }
+
+  get x() {
+    if (this.keys[37] || this.keys[65]) {
```

Comment:
This is tightly coupled with the tank. I think the movement itself is responsibility of the Tank class. This should only give the tank the intention to move in a direction. The movement and the speed of movement will be a property and responsibility of the tank